lib/action_controller/serialization.rb

```diff
@@ -57,11 +57,14 @@ def use_adapter?
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
         options.freeze
-        serializer_options = options.deep_dup
-        serializer_options.fetch(:serialization_context) do
-          serializer_options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, serializer_options)
+        serialization_options = options.deep_dup
+        if options.key?(:serializer)
+          serialization_options[:serializer] = options[:serializer]
```

Comment:
Yes, I still haven't figured it out but there were three bugs failling. The resulting JSON API types looked like `"some_module_some_class"` instead of the values specified via `type "post"`. So I assume when the class was duped the `_type` wasn't duped properly.